miner/worker.go

```diff
@@ -194,10 +204,13 @@ func (w *worker) commitNewWork() (*types.Block, error) {
 }
 
 func (w *worker) createCurrentEnvironment(parent *types.Header, header *types.Header, tstart time.Time) (*environment, error) {
+	// Retrieve the parent state to apply transactions to and kickoff prefetcher,
+	// which will retrieve intermediate nodes for all modified state.
 	state, err := w.chain.StateAt(parent.Root)
 	if err != nil {
 		return nil, err
 	}
+	state.StartPrefetcher("miner", w.eth.BlockChain().CacheConfig().TriePrefetcherParallelism)
```

Comment:
nit: Should we move `StartPrefetcher` to `commitNewWork` right after we call `createCurrentEnvironment`, just to make it close with it's deferred `StopPrefetcher`?

Reply:
See note about mirroring `go-ethereum`: https://github.com/ava-labs/coreth/pull/372#discussion_r1399917719